code/modules/mob/living/carbon/human/human_emote.dm

```diff
@@ -37,6 +37,41 @@
 	emote_type = EMOTE_VISIBLE
 	hands_use_check = TRUE
 
+/datum/emote/living/carbon/human/clap
+	key ="clap"
+	key_third_person = "claps"
+	message = "claps."
+	message_mime = "claps silently."
+	message_param = "claps at %t."
+	emote_type = EMOTE_SOUND
+	vary = TRUE
+
+/datum/emote/living/carbon/human/clap/run_emote(mob/user, params, type_override, intentional)
+	if(!ishuman(user))
+		return
+
+	var/mob/living/carbon/human/H = user
+	if(!H.bodyparts_by_name[BODY_ZONE_L_ARM] || !H.bodyparts_by_name[BODY_ZONE_R_ARM])
+		if(!H.bodyparts_by_name[BODY_ZONE_L_ARM] && !H.bodyparts_by_name[BODY_ZONE_R_ARM])
+			// no arms...
+			to_chat(user, "<span class='warning'>You need arms to be able to clap.</span>")
+		else
+			// well, we've got at least one
+			user.visible_message("[user] makes the sound of one hand clapping.")
+		return TRUE
+
+	return ..()
+
+/datum/emote/living/carbon/human/clap/get_sound(mob/living/user)
+	if(ishuman(user))
```

Comment:
Can you make this an early return? This one's also my fault

Reply:
Fixed with the latest commit!